features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -160,6 +160,26 @@ void ff_memfree(void *p)
 }
 
 // Implementation of diskio functions (see ChaN/diskio.h)
+static WORD disk_get_sector_size(BYTE pdrv)
+{
+    WORD ssize = _ffs[pdrv]->get_erase_size();
```

Comment:
For most of the other block devices an erase is required before programming (flash in particular). The SD card is the exception. If we program less than an erase block on NOR flash, the data will just become corrupted.

For the block device API, the `erase` function makes a block available for programming. The `get_erase_size` should return the smallest block size that can be rewritten. Since the FAT fs doesn't schedule erases seperately, it needs to erase a block during a write. So the sector size here is bounded to the underlying device's `get_erase_size`.

The rounding to 512 is still useful for I2C EEPROM chips, which actually have `get_erase_size` of 1 byte.